### PR TITLE
refactor: split Checklist into sub-components and hooks

### PR DESCRIPTION
**Pull request by jamdelion**
## In this PR:

- Extracted out `GroupedChecklistOptions`, `ExclusiveChecklistOption` and `ChecklistItems` from the `Public` checklist component to make the file easier to read.
- Extracted logic into hooks, e.g. `useExclusiveOption` to move the complexity out of the single dense file.

### CODE REVIEW

File: editor.planx.uk/src/@planx/components/Checklist/Public/Public.tsx [hunk 1]
```diff
@@ -1,40 +1,10 @@
...
-import React, { useState } from "react";
-import { ExpandableList, ExpandableListItem } from "ui/public/ExpandableList";
-import FormWrapper from "ui/public/FormWrapper";
-import FullWidthWrapper from "ui/public/FullWidthWrapper";
-import ChecklistItem from "ui/shared/ChecklistItem/ChecklistItem";
-import ErrorWrapper from "ui/shared/ErrorWrapper";
-import { object } from "yup";
+import React from "react";
 
 import { Option } from "../../shared";
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/@planx/components/Checklist/Public/Public.tsx:4]: nit: Unused import

File: editor.planx.uk/src/@planx/components/Checklist/Public/components/ChecklistItems.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import { Option } from "@planx/components/shared";
+import ImageButton from "@planx/components/shared/Buttons/ImageButton";
+import { FormikProps } from "formik";
+import React from "react";
+import FormWrapper from "ui/public/FormWrapper";
+import ChecklistItem from "ui/shared/ChecklistItem/ChecklistItem";
+
+import { ChecklistLayout } from "./VisibleChecklist";
+
+interface NonExclusiveChecklistItemsProps {
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/@planx/components/Checklist/Public/components/ChecklistItems.tsx:11]: nit: Maybe just `Props` now that this is in isolation?

File: editor.planx.uk/src/@planx/components/Checklist/Public/components/ChecklistItems.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  nonExclusiveOptions: Option[];
+  layout: ChecklistLayout;
+  changeCheckbox: (id: string) => () => void;
+  formik: FormikProps<{ checked: Array<string> }>;
+  exclusiveOptionIsChecked: boolean;
+}
+
+export const NonExclusiveChecklistItems = ({
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/@planx/components/Checklist/Public/components/ChecklistItems.tsx:19]: As per the file name, I actually think that just `ChecklistItems` might be better here.

The combination of `ChecklistItems` and `ExclusiveChecklistItem` still feels meaningful to me.

File: editor.planx.uk/src/@planx/components/Checklist/Public/components/ExclusiveChecklistItem.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  exclusiveOrOption: Option;
+  changeCheckbox: (id: string) => () => void;
+  formik: FormikProps<{ checked: Array<string> }>;
+}) => (
+  <FormWrapper key={exclusiveOrOption.id}>
+    <Grid item xs={12} key={exclusiveOrOption.data.text}>
+      <Typography width={36} display="flex" justifyContent="center">
+        or
+      </Typography>
+
```
Comment (jessicamcinchak) [inline editor.planx.uk/src/@planx/components/Checklist/Public/components/ExclusiveChecklistItem.tsx:24]: nit: I personally do not like empty rows within `return` blocks! Same goes for this line below: https://github.com/theopensystemslab/planx-new/pull/4088/files#diff-b200eeb6b69eea54e2d875c01212d93f0b029698e062e35dcf88c337fedad06dR125

**Review by jessicamcinchak — APPROVED** [review]

File: editor.planx.uk/src/@planx/components/Checklist/Public/hooks/useExclusiveOption.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import { FormikProps } from "formik";
+
+import { Option } from "../../../shared";
+
+export const useExclusiveOption = (
+  exclusiveOptions: Option[],
+  formik: FormikProps<{ checked: Array<string> }>
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/@planx/components/Checklist/Public/hooks/useExclusiveOption.tsx:7]: nit: Seeing this prop many times in this PR points towards `useFormikContext()` being something worth considering. Not a blocker for sure.

**Review by DafyddLlyr — APPROVED** [review]

**Comment by DafyddLlyr** [comment]
A few nitpicks but no show stoppers - nice reorganisation, especially pulling logic out into clear and concise hooks 👍

**Pull request merged**